### PR TITLE
Backpack/ lists

### PR DESCRIPTION
**Pull request by eadams17**
## WHAT
add lists to Backpack

## WHY
we are expanding our component library

## HOW
created new `List` component with 3 style types and add to style guide

### Screenshots
(If applicable. Also, please censor any sensitive data)
Screen Shot 2024-06-03 at 6 58 14 PM

### Notion Card Links
(Please provide links to any relevant Notion card(s) relevant to this PR.)
https://www.notion.so/quill/Quill-Backpack-Buttons-Menus-Text-Fields-Segmented-Controls-Lists-85b0fa6c8cee4a108a015f307befb7a7?pvs=4

### What have you done to QA this feature?
testing on staging that the different lists behave and display as expected

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  yes
Have you deployed to Staging? | yes
Self-Review: Have you done an initial self-review of the code below on Github? | yes

### CODE REVIEW

File: services/QuillLMS/client/app/bundles/Shared/components/shared/list.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+const SECONDARY = 'secondary'
+const CHECKBOX = 'checkbox'
+
+export const List = ({ items, style }: ListProps) => {
+  if(style === PRIMARY) {
+    return(
+      <ul className="list quill-list single-line">
+        {items.map(({ label, onClick }) => (
+          <li>
+            <button className="interactive-wrapper" onClick={onClick}>{label}</button>
```
Comment (emilia-friedberg) [inline services/QuillLMS/client/app/bundles/Shared/components/shared/list.tsx]: I think this button probably needs a `type` to pass the linter, right? I also wonder if we want to give it a `focus-on-light` class.

File: services/QuillLMS/client/app/bundles/Shared/components/shared/list.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        ))}
+      </ul>
+    )
+  }
+  if(style === SECONDARY) {
+    return (
+      <ul className="list quill-list double-line">
+        {items.map(({ primaryText, secondaryText, onClick }) => (
+          <li>
+            <button className="interactive-wrapper" onClick={onClick}>
```
Comment (emilia-friedberg) [inline services/QuillLMS/client/app/bundles/Shared/components/shared/list.tsx]: See comment on button above.

**Review by emilia-friedberg — COMMENTED** [review]
I left a couple of notes about the buttons, but more generally, I'm wondering if there is always on `onClick` handler getting passed in here? Maybe I'm wrong but the design comp to me looks like maybe sometimes it's just a static list, in which case I think we want to do some logic to check to see if an `onClick` is getting passed in and if not, render a static element rather than a button (to avoid weird UX for keyboard users and others).

File: services/QuillLMS/client/app/bundles/Shared/components/shared/list.tsx (hunk 1 above)
Comment (eadams17) [reply services/QuillLMS/client/app/bundles/Shared/components/shared/list.tsx]: I think that's a reasonable approach-- I just assumed by the focus and active states that these would always be interactive elements. I'll add logic to check for a click handler and if none is present, just render a regular `li` element

**Review by emilia-friedberg — APPROVED** [review]
Looks great! Thank you for making that change.

**Pull request merged**